src/js/components/TableCell/TableCell.js

```diff
@@ -125,31 +125,48 @@ const TableCell = forwardRef(
       );
     }
 
+    const themeProviderValue = { ...theme };
+
+    if (cellProps.background || theme.darkChanged) {
+      const dark = backgroundIsDark(cellProps.background, theme);
+      const darkChanged = dark !== undefined && dark !== theme.dark;
+      if (darkChanged || theme.darkChanged) {
+        themeProviderValue.dark = dark === undefined ? theme.dark : dark;
+        themeProviderValue.background = cellProps.background;
+      } else if (cellProps.background) {
+        // This allows DataTable to intelligently set the background of a pinned
+        // header or footer.
+        themeProviderValue.background = cellProps.background;
+      }
+    }
+
     return (
-      <StyledTableCell
-        ref={cellRef}
-        as={scope ? 'th' : undefined}
-        scope={scope}
-        size={size}
-        colSpan={colSpan}
-        tableContext={tableContext}
-        tableContextTheme={tableContextTheme}
-        {...(plain === true ? mergedProps : {})}
-        {...cellProps}
-        className={className}
-      >
-        {plain || !Object.keys(mergedProps).length ? (
-          content
-        ) : (
-          <Box
-            {...mergedProps}
-            align={align}
-            justify={verticalAlignToJustify[verticalAlign]}
-          >
-            {children}
-          </Box>
-        )}
-      </StyledTableCell>
+      <ThemeContext.Provider value={themeProviderValue}>
```

Comment:
I think this is OK. I was trying to figure out if there are any issues with themeProviderValue basically changing each render. Probably not and it's good that it always includes this context so we don't run into what we saw in #5146

Reply:
Good thoughts. I've reworked the code to avoid redoing the ThemeContext unless there's something to put in it.